### PR TITLE
test(planner): add predicata push down planner test

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/frontend/planner_test/tests/testdata/predicate_pushdown.yaml [hunk 1]
```diff
@@ -58,11 +58,208 @@
     LogicalFilter { predicate: (min(t.v2) > 1:Int32) AND (t.v1 > min(t.v2)) }
     └─LogicalAgg { group_key: [t.v1], aggs: [min(t.v2)] }
       └─LogicalScan { table: t, output_columns: [t.v1, t.v2], required_columns: [v1, v2], predicate: (t.v1 > 1:Int32) AND (1:Int32 > 0:Int32) }
-- name: Always false should not be pushed below SimpleAgg
```
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/predicate_pushdown.yaml:61]: Why to delete this test case?

**Review by xxchan — APPROVED** [review]
I think part of the new test cases, e.g., filter join are already covered by existing ones.. Maybe delete duplicates.

File: src/frontend/planner_test/tests/testdata/predicate_pushdown.yaml [hunk 2]
```diff
@@ -58,11 +58,208 @@
...
+    LogicalProject { exprs: [t.v1, t.v2, t.v3, t.v4] }
+    └─LogicalFilter { predicate: (t.v1 = 10:Int32) AND (t.v2 = 20:Int32) AND (t.v3 = 30:Int32) }
+      └─LogicalTopN { order: "[t.v1 ASC]", limit: 10, offset: 0 }
+        └─LogicalProject { exprs: [t.v1, t.v2, t.v3, t.v4] }
+          └─LogicalScan { table: t, columns: [t.v1, t.v2, t.v3, t.v4, t._row_id] }
+  optimized_logical_plan: |
+    LogicalFilter { predicate: (t.v1 = 10:Int32) AND (t.v2 = 20:Int32) AND (t.v3 = 30:Int32) }
+    └─LogicalTopN { order: "[t.v1 ASC]", limit: 10, offset: 0 }
+      └─LogicalScan { table: t, columns: [t.v1, t.v2, t.v3, t.v4] }
+- name: filter hop transpose
```
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/predicate_pushdown.yaml:89]: Not implemented? Maybe add a todo

File: src/frontend/planner_test/tests/testdata/predicate_pushdown.yaml (hunk 1 above)
Comment (st1page) [reply src/frontend/planner_test/tests/testdata/predicate_pushdown.yaml:61]: by mistake. fixed.

**Comment by st1page** [comment]
Sorry for that I was in a hurry to get off work and I should be more careful. :smiling_face_with_tear:

**Pull request merged**